### DISCUSSION TITLE
EvalauteAndLog Not Showing from Power BI Desktop

### DISCUSSION THREAD

**Discussion by trevrobwhite** [DaxStudio/DaxStudio#1307] [Q&A]
DaxStudio is an awesome tool for helping me with Power BI, but there is one feature I'm struggling to get working, that is EvalauteAndLog, I've gone through the [documentation](https://daxstudio.org/docs/tutorials/evaluateandlog-trace/) and I can see the output in DaxDebugOutput application, if I close that app and reopen DaxStudio, set Evaluate & Log to record nothing appears even when I change the measure in Power BI desktop.

If I run EvalauteAndLog directly in Dax Studio then it appears but not Power BI (it's connected and the latest version).

I also tried opening Dax Studio from the External tools button in Power BI.

Any ideas on what I can try?

**Comment by marcosqlbi** [comment]
You should:
1) Capture the query from the visual in PBI Desktop using Performance Analyzer
2) Run the query on DAX Studio

I hope it helps!

**Reply by trevrobwhite** [reply]
Ah ok, it doesn't grab it from Power BI direct like DaxDebugOutput, would be a great feature if it did at some point to remove the need for the second tool :-)

**Reply by marcosqlbi** [reply]
It would be expensive - not impossible. @dgosbell what do you think? Could it be something related to All Queries?

**Reply by dgosbell** [reply]
The current design of the EvaluateAndLog trace is that it only "listens" to events coming from the session for the DAX Studio window it's running in. The idea here is that you can use the Query Builder and override your measure expression to include the EvalueateAndLog() calls or you can use [Define Measure](https://daxstudio.org/docs/features/define-measure/) to override your measure expression. This just affects the current query scope without having to save any of the EvaluateAndLog() calls into your pbix file. In this way you don't have to jump back and forth between different tools, you can do all your debugging in DAX Studio and then only transfer the final measure expression back to your model once you are happy with it.

If you read through the blog post from one of the core developers of the EvaluateAndLog function here https://pbidax.wordpress.com/2022/08/16/introduce-the-dax-evaluateandlog-function/ you will see in the last 3 paragraphs that he talks about the fact that this function has to disable a number of internal engine optimizations in order for it to work, so he does not recommend leaving in code that might be deployed to production.

For this reason I'm hesitant to change the feature to make it "listen" to events from all sessions. There is also the problem that if you have multiple visuals and multiple measures with EvaluateAndLog it could be tricky to figure out which events were raised by which measure unless you are very careful with your labelling.

**Comment by trevrobwhite** [comment]
That makes total sense the issue in my case was a problem with a visual not loading due to lack of resources so I was using the other tool to work out the filters causing the problem that were affecting the measure.

If it’s a simple change, could it maybe an option which events to listen for from PowerBI?

Thank you again for the super detail and ultra quick reply.
________________________________

**Reply by marcosqlbi** [reply]
If you have a resources issue, the last thing you want to do is to enable EvaluateAndLog. It would be best if you analyzed Server Timings and Query Plan to figure out what is draining memory.
